File: 07_app_development/MyProject/libtsparser/src/main/java/com/tosmart/xiexuming/libtsparser/descriptor/ServiceDescriptor.java
package com.tosmart.xiexuming.libtsparser.descriptor;

import androidx.annotation.NonNull;


/**
 * @author xiexuming
 */
public class ServiceDescriptor extends Descriptor {
    private int serviceType;
    private int serviceProviderNameLength;
    private String serviceProviderName;
    private int serviceNameLength;
    private String serviceName;

    @NonNull
    @Override
    public String toString() {
        return "ServiceDescriptor{" +
                "descriptorTag=" + descriptorTag +
                ", descriptorLength=" + descriptorLength +
                ", serviceType=" + serviceType +
                ", serviceProviderNameLength=" + serviceProviderNameLength +
                ", serviceProviderName=" + serviceProviderName +
                ", serviceNameLength=" + serviceNameLength +
                ", serviceName=" + serviceName +
                '}';
    }

    public int getServiceType() {
        return serviceType;
    }

    public void setServiceType(int serviceType) {
        this.serviceType = serviceType;
    }

    public int getServiceProviderNameLength() {
        return serviceProviderNameLength;
    }

    public void setServiceProviderNameLength(int serviceProviderNameLength) {
        this.serviceProviderNameLength = serviceProviderNameLength;
    }

    public int getServiceNameLength() {
        return serviceNameLength;
    }

    public void setServiceNameLength(int serviceNameLength) {
        this.serviceNameLength = serviceNameLength;
    }

    public String getServiceProviderName() {
        return serviceProviderName;
    }

    public void setServiceProviderName(String serviceProviderName) {
        this.serviceProviderName = serviceProviderName;
    }

    public String getServiceName() {
        return serviceName;
    }

    public void setServiceName(String serviceName) {
        this.serviceName = serviceName;
    }
}

File: newpaserapp/libtosmartparse/src/main/java/com/example/libtosmartparse/descriptor/ServiceDescriptor.kt
package com.example.libtosmartparse.descriptor


/**
 * @author xxm
 * 2022/7/17
 **/
class ServiceDescriptor : Descriptor() {
    private var serviceType = 0
    private var serviceProviderNameLength = 0
    private var serviceProviderName: String? = null
    private var serviceNameLength = 0
    private var serviceName: String? = null

    override fun toString(): String {
        return "ServiceDescriptor{" +
                "descriptorTag=" + getDescriptorTag() +
                ", descriptorLength=" + getDescriptorLength() +
                ", serviceType=" + serviceType +
                ", serviceProviderNameLength=" + serviceProviderNameLength +
                ", serviceProviderName=" + serviceProviderName +
                ", serviceNameLength=" + serviceNameLength +
                ", serviceName=" + serviceName +
                '}'
    }

    fun getServiceType(): Int {
        return serviceType
    }

    fun setServiceType(serviceType: Byte) {
        this.serviceType = serviceType.toInt()
    }

    fun getServiceProviderNameLength(): Int {
        return serviceProviderNameLength
    }

    fun setServiceProviderNameLength(serviceProviderNameLength: Int) {
        this.serviceProviderNameLength = serviceProviderNameLength
    }

    fun getServiceNameLength(): Int {
        return serviceNameLength
    }

    fun setServiceNameLength(serviceNameLength: Int) {
        this.serviceNameLength = serviceNameLength
    }

    fun getServiceProviderName(): String? {
        return serviceProviderName
    }

    fun setServiceProviderName(serviceProviderName: String?) {
        this.serviceProviderName = serviceProviderName
    }

    fun getServiceName(): String? {
        return serviceName
    }

    fun setServiceName(serviceName: String?) {
        this.serviceName = serviceName
    }
}
